src/pages/Home.jsx

```diff
@@ -1,12 +1,43 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import countriesData from '../assets/CountriesData.json'
+import Country from '../components/Country'
+import Filter from "../components/Filter"
+const Home = (props) => {
+    const [searchValue, setSearchValue] = useState("")
+    const [selectedRegion, setSelectedRegion] = useState("All")
+    const [countries, setCountries] = useState(countriesData)
+    const [currentCountriesAfterRegion, setCurrentCountriesAfterRegion] = useState(countriesData)
+    useEffect(() => {
+        let data = {}
+        if (selectedRegion != "All") {
+            data = countriesData.filter((country) => country.region.toLowerCase() == selectedRegion.toLowerCase())
+        }
+        else data = countriesData
+        setCurrentCountriesAfterRegion(data)
+        setCountries(data)
+    }, [selectedRegion])
 
-const Home = () => {
+    useEffect(() => {
+        const data = currentCountriesAfterRegion.filter((country) => country.name.toLowerCase().includes(searchValue.toLowerCase()))
+        setCountries(data)
```

Comment:
A question, do you use `useEffect` in here to prevent re calculating the filter logic again and again? or for something else

Reply:
I want that once the user is looking for a country to make that filter and no more then that, that's why I put in the function call on the second param the [searchValue], so once it get changed, the useEffect get started